src/components/draw/DrawDirective.js

```diff
@@ -161,165 +317,228 @@ goog.require('ga_map_service');
 
             // Remove interactions
             deactivateDrawInteraction();
-            deactivateSelectInteraction();
-            deactivateModifyInteraction();
+            select.setActive(false);
+
+            // Unregister the events attached to the source
+            for (var i in unSourceEvents) {
+              ol.Observable.unByKey(unSourceEvents[i]);
+            }
+            unSourceEvents = [];
+
           };
 
-          // Deactivate other tools
-          var activateTool = function(tool) {
-            layer.visible = true;
+          // Set the draw interaction with the good geometry
+          var deregDrawStart, deregDrawEnd, deregPointerMove2;
```

Comment:
Those 3 dereg variables could be replace by a single array. Then push to this array when creating them, and use array forEach to dereg.